### DISCUSSION TITLE
ics-py can run in the browser using JypyterLite

### DISCUSSION THREAD

**Discussion by introt** [ics-py/ics-py#372] [Show and tell]
> [JupyterLite](https://jupyterlite.readthedocs.io/en/latest/) is all about accessible browser-based interactive computing: Python kernel backed by Pyodide running in a Web Worker

![image](https://user-images.githubusercontent.com/26069787/204637707-758cc50a-33eb-4518-b147-1f1968b4a2c2.png)

You can also install the latest development version by substituting `ics` with `ics==0.8.0.dev0`. It's also possible to skip this step by bundling packets and files with your deployment of JupyterLite.

**Comment by C4ptainCrunch** [comment]
This is awesome :)
We should maybe try one day to integrate this in the docs, it would be nice for the users to have a sandbox to play with ics in their browser.

**Reply by introt** [reply]
Yes, I was thinking about interactive examples/doctests - not sure whether it'd be easier to generate Jupyter Notebooks from docs or the other way around. I'll be trying this out myself with [zim2anki](https://github.com/introt/zim2anki) as soon as I get that thing off the ground - sadly sockets are not supported.

**Reply by N-Coder** [reply]
FYI [`jupytext`](https://github.com/mwouts/jupytext/#readme) very nicely provides a conversation between markdown files with python snippets to notebooks with markdown and code cells.

**Reply by introt** [reply]
Have you come across similar tools for converting docstrings + doctests or unittests with docstrings into jupyter notebooks? Interactive [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) docs would be pretty cool imo.

**Reply by N-Coder** [reply]
It should be pretty easy to get a suitable intermediate representation via sphinx or by using python runtime introspection. But I'm not 100% convinced this is really something we want, as it might motivate cramming as much as possible into in-code docstrings. I'd prefer having comprehensive (interactive) examples interleaved with their explanations and not their underlying implementations, so putting all this into the (markdown/rst) documentation files and converting those directly to notebooks might be better than making docstrings that central.

**Reply by introt** [reply]
Yes, I get your point. Practicality aside, I'd be interested to try some "doctest driven development" in a notebook on a side project for fun :) Why stick to static docs or [blogposts(!)](https://github.com/ubuntu/zsys#documentation) when you can explain your code/API/decisions with interactive inline ixamples?
